### PR TITLE
Add test cases for ansi

### PR DESCRIPTION
**Pull request by kiyonlin**
Hey @muesli I'm a fan of `bubbletea` now 😄  Thanks for your awesome work!

And I found `reflow` have room for performance improvement and test coverage. 

I plan to improve them step by step.

This PR only added test cases for asni. If you accept it, I'll do the performance improvement work.

### CODE REVIEW

File: ansi/writer_test.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+type fakeWriter struct{}
+
+func (fakeWriter) Write(_ []byte) (int, error) {
+	return 0, errors.New("fake error")
+}
+
+func TestWriter_Write_Error(t *testing.T) {
+
```
Comment (muesli) [inline ansi/writer_test.go:40]: Spurious newline.

File: ansi/writer_test.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	w := &Writer{Forward: fakeWriter{}}
+
+	_, err := w.Write([]byte("foo"))
+
+	if err == nil {
+		t.Fatal("err shouldn't be nil")
```
Comment (muesli) [inline ansi/writer_test.go:46]: We should probably print out the error that we got instead of `nil`.

File: ansi/writer_test.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	)
+	b.ReportAllocs()
+	b.ResetTimer()
+
+	for i := 0; i < b.N; i++ {
+		n, err = w.Write(buf)
+	}
+
+	if err != nil {
+		b.Fatal("err should be nil")
```
Comment (muesli) [inline ansi/writer_test.go:65]: We should probably print out the error that we got instead of nil.

File: ansi/writer_test.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+	w := &Writer{Forward: ioutil.Discard}
+
+	n, err := w.Write(buf)
+
+	w.ResetAnsi()
+
+	w.RestoreAnsi()
+
+	if err != nil {
+		t.Fatal("err should be nil")
```
Comment (muesli) [inline ansi/writer_test.go:21]: We should probably print out the error that we got instead of nil.

**Review by muesli — CHANGES_REQUESTED** [review]
Sorry for the delayed review @kiyonlin, the last few days have been super hectic :smiley:

Looks good to me, just a few nitpicky change requests. Thank you for contributing!

**Comment by kiyonlin** [comment]
Thanks for your detail review 😄 

> Sorry for the delayed review @kiyonlin, the last few days have been super hectic 😃

You guys are famous now 👍

**Pull request merged**